ml-agents/mlagents/trainers/tf_policy.py

```diff
@@ -217,22 +218,12 @@ def export_model(self):
 
         with self.graph.as_default():
             target_nodes = ",".join(self._process_graph())
-            ckpt = tf.train.get_checkpoint_state(self.model_path)
-            freeze_graph.freeze_graph(
-                input_graph=self.model_path + "/raw_graph_def.pb",
-                input_binary=True,
-                input_checkpoint=ckpt.model_checkpoint_path,
-                output_node_names=target_nodes,
-                output_graph=(self.model_path + "/frozen_graph_def.pb"),
-                clear_devices=True,
-                initializer_nodes="",
-                input_saver="",
-                restore_op_name="save/restore_all",
-                filename_tensor_name="save/Const:0",
+            graph_def = self.graph.as_graph_def()
+            output_graph_def = graph_util.convert_variables_to_constants(
+                self.sess, graph_def, target_nodes.replace(" ", "").split(",")
             )
-
-        tf2bc.convert(self.model_path + "/frozen_graph_def.pb", self.model_path + ".nn")
-        logger.info("Exported " + self.model_path + ".nn file")
+            tf2bc.convert_graph_def(output_graph_def, self.model_path + ".nn")
```

Comment:
It is not clear to me what is a graph def. I thought it was just the "skeleton" of the graph (without any weights on it). Here it seems to imply that using the graph def alone is enough to create the `.nn` file.

Reply:
Specifically this script manages graph defs where all variables are converted to constants.  Seems like the GraphDef can hold values as well.

Reply:
This is confusing. Can we maybe rename it to something like `graph_def_with_constant_weights`?

Reply:
Based on Mantas' feedback, I've removed this change to tf2bc and returned to outputting the frozen graph to a file.